packages/caver-account/src/accountKey/accountKeyRoleBased.js

```diff
@@ -0,0 +1,144 @@
+/*
+    Copyright 2020 The caver-js Authors
+    This file is part of the caver-js library.
+
+    The caver-js library is free software: you can redistribute it and/or modify
+    it under the terms of the GNU Lesser General Public License as published by
+    the Free Software Foundation, either version 3 of the License, or
+    (at your option) any later version.
+
+    The caver-js library is distributed in the hope that it will be useful,
+    but WITHOUT ANY WARRANTY; without even the implied warranty of
+    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+    GNU Lesser General Public License for more details.
+
+    You should have received a copy of the GNU Lesser General Public License
+    along with the caver-js. If not, see <http://www.gnu.org/licenses/>.
+*/
+
+const RLP = require('eth-lib/lib/rlp')
+const AccountKeyWeightedMultiSig = require('./accountKeyWeightedMultiSig')
+const AccountKeyLegacy = require('./accountKeyLegacy')
+const AccountKeyPublic = require('./accountKeyPublic')
+const AccountKeyFail = require('./accountKeyFail')
+const utils = require('../../../caver-utils')
+const { ACCOUNT_KEY_TAG } = require('./accountKeyHelper')
+
+function isValidAccountKeysFormat(roleBasedAccountKeys) {
```

Comment:
This function assumes that the parameter is a role-based account key. What if another account key type is passed?
If this assumption is right, I think the function name should be changed to describe such assumption.

Reply:
This is a function that verifies that the format is correct before assigning it to the accountKey variable.

roleBasedAccountKeys parameter should be in the form of an array of AccountKey(AccountKeyLegacy, Public, Fail, WeightedMultiSig) instances, and to check this, i added logic to check if it is an array.

So, if another accountKey comes in, it returns false because that cannot be assigned to accountKey variable.